catalog/app/constants/routes.ts

```diff
@@ -133,17 +133,25 @@ export const bucketPackageList = route(
 )
 export type BucketPackageListArgs = Parameters<typeof bucketPackageList.url>
 
-interface BucketPackageDetailOpts {
-  action?: string
-}
-
 export const bucketPackageDetail = route(
   `/b/:bucket/packages/:name(${PACKAGE_PATTERN})`,
-  (bucket: string, name: string, { action }: BucketPackageDetailOpts = {}) =>
-    `/b/${bucket}/packages/${name}${mkSearch({ action })}`,
+  (bucket: string, name: string) => `/b/${bucket}/packages/${name}`,
 )
+
 export type BucketPackageDetailArgs = Parameters<typeof bucketPackageDetail.url>
 
+interface BucketPackageAddFilesOpts {
+  [logicalKey: string]: string // S3 url
+}
+
+export const bucketPackageAddFiles = route(
+  `/b/:bucket/packages/:name(${PACKAGE_PATTERN})/add-files`,
+  (bucket: string, name: string, files: BucketPackageAddFilesOpts = {}) =>
+    `/b/${bucket}/packages/${name}/add/${mkSearch(files)}`,
+)
```

Comment:
1. "forward" and "backward" urls don't match -- one uses `/add-files` trailing segment, and the other uses `/add/`. should be the same.
2. i don't quite get why it's necessary to have a separate route for this. why can't you use a single route with optional params/segments? i suspect this would also cause some unnecessary re-renders and state loss while navigating between those, since the downstream component (PackageTree) will be recreated with fresh state.

Reply:
1. It's a first step to have a separate page for creating packages
2. The new URL contains files as a query string: `?file1=s3://file1&file2=s3://file2`. So, if we keep previous format ("`package-url?action=revisePackage`"), it would conflict with files